network/common/src/main/java/org/apache/spark/network/server/StreamManager.java

```diff
@@ -44,6 +46,17 @@
   public abstract ManagedBuffer getChunk(long streamId, int chunkIndex);
 
   /**
+   * Register the given stream to the associated channel. So these streams can be cleaned up later.
```

Comment:
Let's beef up the documentation:

Associates a stream with a single client connection, which is guaranteed to be the only reader of the stream. The getChunk() method will be called serially on this connection and once the connection is closed, the stream will never be used again, enabling cleanup.

This must be called before the first getChunk() on the stream, but it may be invoked multiple times with the same channel and stream id.